src/rosdiscover/recover/states_analyzer.py

```diff
@@ -41,10 +43,28 @@ def state_vars(self) -> t.List[SymbolicVariableReference]:
                         var_refs.append(expr)
         return var_refs
 
+    @cached_property
+    def state_vars(self) -> t.List[SymbolicVariableReference]:
+        result: t.List[SymbolicVariableReference] = []
+        for var in self._state_vars:
+            if var.initial_value.is_unknown():
```

Comment:
suggest: `if not var.initial_value.is_unknown():\ncontinue` to reduce nesting/complexity